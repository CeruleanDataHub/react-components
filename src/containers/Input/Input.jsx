import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

const InputWrapper = styled.input`
  padding: 1rem;
  width: 100%;
`;

export const Input = ({ type, placeholder, onChange }) => (
  <InputWrapper type={type} onChange={onChange} placeholder={placeholder} />
);

Input.propTypes = {
  /** Input type */
  type: PropTypes.oneOf(["text", "search"]),
  /** Placeholder text */
  placeholder: PropTypes.string,
  /** onChangehandler function */
  onChange: PropTypes.func
};

Input.defaultProps = {
  type: "text",
  placeholder: "",
  onChange: () => {}
};
