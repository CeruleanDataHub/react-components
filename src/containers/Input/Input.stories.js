import { action } from "@storybook/addon-actions";
import { withKnobs } from "@storybook/addon-knobs";
import React from "react";

import { Input } from "./Input";

export default {
  title: "Input",
  component: Input,
  decorators: [withKnobs]
};

export const InputStory = () => (
  <Input placeholder="Write your thing here" onChange={action("on change")} />
);

export const InputSearchStory = () => (
  <Input
    type="search"
    placeholder="Write your thing here"
    onChange={action("on change")}
  />
);
