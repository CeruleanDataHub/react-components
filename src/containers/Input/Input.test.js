import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";

import { Input } from "./Input";

describe("Input", () => {
  it("should render", () => {
    const component = renderer.create(<Input />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render placeholder", () => {
    const component = renderer.create(<Input placeholder="test" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should be able to call onChange function", () => {
    const handleOnChange = jest.fn();
    const wrapper = shallow(
      <Input placeholder="test" onChange={handleOnChange} />
    );

    wrapper.simulate("change", "test");
    expect(handleOnChange).toHaveBeenCalledWith("test");
  });

  it("should be able to call default onChange function when no function is passed", () => {
    const component = renderer.create(<Input placeholder="test" />);
    const tree = component.toJSON();
    component.root.props.onChange();
    expect(tree).toMatchSnapshot();
  });

  describe("types", () => {
    it("should render text input", () => {
      const component = renderer.create(
        <Input type="text" placeholder="test" />
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should render search input", () => {
      const component = renderer.create(
        <Input type="search" placeholder="test" />
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
