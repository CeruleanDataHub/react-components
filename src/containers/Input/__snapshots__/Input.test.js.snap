// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should be able to call default onChange function when no function is passed 1`] = `
.c0 {
  padding: 1rem;
  width: 100%;
}

<input
  className="c0"
  onChange={[Function]}
  placeholder="test"
  type="text"
/>
`;

exports[`Input should render 1`] = `
.c0 {
  padding: 1rem;
  width: 100%;
}

<input
  className="c0"
  onChange={[Function]}
  placeholder=""
  type="text"
/>
`;

exports[`Input should render placeholder 1`] = `
.c0 {
  padding: 1rem;
  width: 100%;
}

<input
  className="c0"
  onChange={[Function]}
  placeholder="test"
  type="text"
/>
`;

exports[`Input types should render search input 1`] = `
.c0 {
  padding: 1rem;
  width: 100%;
}

<input
  className="c0"
  onChange={[Function]}
  placeholder="test"
  type="search"
/>
`;

exports[`Input types should render text input 1`] = `
.c0 {
  padding: 1rem;
  width: 100%;
}

<input
  className="c0"
  onChange={[Function]}
  placeholder="test"
  type="text"
/>
`;
